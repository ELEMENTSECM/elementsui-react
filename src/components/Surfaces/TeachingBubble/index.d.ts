import * as React from 'react';

export interface TeachingBubbleProps {
	/**
	 * Teachingbubble Headline
	 */
	headline?: string;
	/**
	 * Teachingbubble Target
	 */
	targetElement?: Object;
	/**
	 * Teachingbubble Dissmissed
	 */
	dismissed?: boolean;
	/**
	 * isTeachingBubbleVisible
	 */
	isTeachingBubbleVisible?: boolean;
}

export const TeachingBubble: React.SFC<TeachingBubbleProps>;
