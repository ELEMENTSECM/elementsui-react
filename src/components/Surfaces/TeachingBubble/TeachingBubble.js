import React from 'react';
import PropTypes from 'prop-types';
import { TeachingBubble as UIFabTeachingBubble } from 'office-ui-fabric-react/lib/TeachingBubble';
import { getStyles } from './TeachingBubble.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** TeachingBubble example */
export function TeachingBubble(props) {
	const {
		headline,
		targetElement,
		dismissed,
		isTeachingBubbleVisible,
		element,
		className,
		theme,
		getStyles
	} = props;
	const classNames = classNamesFunction()(getStyles, props);
	return (
		<div>
			<UIFabTeachingBubble
				className={classNames.root}
				headline={headline}
				targetElement={targetElement}
				dismissed={dismissed}
				isTeachingBubbleVisible={isTeachingBubbleVisible}>
				{element}
			</UIFabTeachingBubble>
		</div>
	);
}

TeachingBubble.propTypes = {
	/**Teachingbubble Headline */
	headline: PropTypes.string,
	/**Teachingbubble Target */
	targetElement: PropTypes.object,
	/**Teachingbubble Dissmissed */
	dismissed: PropTypes.bool,
	/**isTeachingBubbleVisible */
	isTeachingBubbleVisible: PropTypes.bool
};

export default styled(customizable('TeachingBubble', ['theme'])(TeachingBubble), getStyles);
