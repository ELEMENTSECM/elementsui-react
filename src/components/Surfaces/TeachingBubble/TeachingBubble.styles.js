export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-TeachingBubble',
			{
				// background: props.theme.palette.themePrimary
				// place your styles here
			},
			className
		]
	};
};
