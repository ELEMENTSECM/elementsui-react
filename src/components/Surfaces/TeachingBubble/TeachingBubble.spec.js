import React from 'react';
import renderer from 'react-test-renderer';
import TeachingBubble from './TeachingBubble';

describe('TeachingBubble', () => {
	test('displays `TeachingBubble` as headline', () => {
		const tree = renderer
			.create(<TeachingBubble htmlId="defaultTeachingBubble" headline="TeachingBubble" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
