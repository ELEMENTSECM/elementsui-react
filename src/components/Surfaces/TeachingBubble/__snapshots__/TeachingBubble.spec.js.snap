// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeachingBubble displays \`TeachingBubble\` as headline 1`] = `
<div>
  <span
    className="ms-Layer"
  />
</div>
`;
